ui/js/util/form-validation.js

```diff
@@ -0,0 +1,35 @@
+// these don't need to be exact
+// shapeshift does a more thourough check on validity
+// just general matches to prevent unneccesary api calls
+export const coinRegexPatterns = {
+  BTC: /^[13][a-km-zA-HJ-NP-Z1-9]{25,34}$/,
+  BCH: /^[13][a-km-zA-HJ-NP-Z1-9]{25,34}$/,
+  ETH: /^(0x)?[0-9a-fA-F]{40}$/,
+  DASH: /^X([0-9a-zA-Z]){33}/,
+  LTC: /^[a-zA-Z0-9]{26,33}$/,
+  XMR: /^4[0-9ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz]{94}$/,
+};
+
+const validateAddress = (coinType, address) => {
```

Comment:
Great work starting a validation library! Let's determine a consistent signature for validation functions.

One that I've seen and like is that validators return sanitized values and except if they encounter an error.

So far, this is two inconsistent signatures, one that returns true/false, and one that returns an error set.